tools/stress-cluster/scripts/deploy_stress_tests.ps1

```diff
@@ -0,0 +1,134 @@
+[CmdletBinding(DefaultParameterSetName = 'Default')]
+param(
+    [string]$searchDirectory,
+    [hashtable]$filters,
+    [string]$environment,
+    [string]$uploadToRegistry,
+    [string]$repository,
+
+    [Parameter(Mandatory=$true)]
+    [string]$deployId,
+
+    [Parameter(ParameterSetName = 'DoLogin', Mandatory = $true)]
+    [string]$subscription,
+
+    [Parameter(ParameterSetName = 'DoLogin', Mandatory = $true)]
+    [string]$clusterName,
+
+    [Parameter(ParameterSetName = 'DoLogin', Mandatory = $true)]
+    [string]$clusterGroup
+)
+
+. $PSScriptRoot/find_all_stress_packages.ps1
+
+# Powershell does not (at time of writing) treat exit codes from external binaries
+# as cause for stopping execution, so do this via a wrapper function.
+# See https://github.com/PowerShell/PowerShell-RFC/pull/277
+function run() {
+    Write-Output "" "==> $args" ""
+    $command, $arguments = $args
+    & $command $arguments
+    if (!$?) {
+        $code = $LASTEXITCODE
+        Write-Output "Command '$args' failed with code: $code"
+        exit $code
+    }
+}
+
+$runFunctionInit=[scriptblock]::create(@"
+function run() {
+    $function:run
+}
+"@)
+
+function login([string]$subscription, [string]$clusterName, [string]$clusterGroup, [string]$uploadToRegistry) {
+    Write-Output "Logging in to subscription, cluster and container registry"
+    az account show
+    if (!$?) {
+        run az login --allow-no-subscriptions
+    }
+
+    run az aks get-credentials `
+        -n "$clusterName" `
+        -g "$clusterGroup" `
+        --subscription "$subscription" `
+        --overwrite-existing
+
+    if ($uploadToRegistry) {
+        run az acr login -n $uploadToRegistry
+    }
+}
+
+function deployStressTests(
+    [string]$searchDirectory = '.',
+    [hashtable]$filters = @{},
+    [string]$environment = 'test',
+    [string]$uploadToRegistry,
+    [string]$repository = 'images',
+    [string]$deployId,
+    [string]$subscription,
+    [string]$clusterName,
+    [string]$clusterGroup
+) {
+    if ($PSCmdlet.ParameterSetName -eq 'DoLogin') {
+        login $subscription $clusterName $clusterGroup $uploadToRegistry
+    }
+
+    run helm repo add stress-test-charts https://stresstestcharts.blob.core.windows.net/helm/
+    run helm repo update
+
+    findStressPackages $searchDirectory $filters | % {
+        $args = $_, $deployId, $environment, $uploadToRegistry, $repository
```

Comment:
While fill in args like this as opposed to just passing them to the function below?

Reply:
Another side effect of populating an `ArgumentList` for jobs that can now be removed.